src/utils.jl

```diff
@@ -89,3 +89,48 @@ function optional(ex::Expr)
 
     return funcs
 end
+
+
+function _gen_replace(T::Type, struct_var::Symbol, pair_var::Symbol)
```

Comment:
My hunch is that this is not faster than iterating over the fields in a normal function, which would be much more readable.